### DISCUSSION TITLE
MAIN and DEV branch using GitHub workflow vs. Production and Development environment

### DISCUSSION THREAD

**Discussion by southernsun** [jasontaylordev/CleanArchitecture#1159] [Q&A] [answered]
Hi Jason/community,

I was wondering if i could get some clarification behind the idea for the current GitHub workflow setup? Why is there only a deployment for the main branch, and does that deploy a development, staging and production release. Won't it make more sense to link these to separate branches: dev->staging->main ? Is there a particular reason for the current setup that i'm perhaps missing? some thought or rational? Just curios, i will probably update the GitHub workflow files to do as described here, using a dev, staging and production branch. Thanks!

**Comment by jasontaylordev** [comment, marked as answer]
Great question! The current GitHub workflow setup in the template uses a trunk-based branching strategy, which is why there's only a deployment for the main branch. In trunk-based development, the main branch serves as the central source of truth, and features are typically developed in short-lived branches that are frequently merged back into the main branch. This approach aims to reduce integration issues and streamline the development process. You can read more about trunk-based development and its benefits [here](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development).

**Reply by southernsun** [reply]
thanks, i will update the Deployment wiki page later today/tomorrow to include this information. Might be useful for future users. :-)

**Reply by jasontaylordev** [reply]
Thanks for offering to update the wiki! I've created a new ADR for our branching strategy: [ADR 002 Adopting Trunk‐Based Development for Branching Strategy](https://github.com/jasontaylordev/CleanArchitecture/wiki/ADR-002-Adopting-Trunk%E2%80%90Based-Development-for-Branching-Strategy). If you could detail the decision there, that would be fantastic!

**Reply by southernsun** [reply]
I added some general comments and text (using ChatGPT). I think it is a nice start and highlights the main pro's and con's for TBD.

**Reply by jasontaylordev** [reply]
That's great, thank you! 😀
